src/portable/stage1.rs

```diff
@@ -0,0 +1,195 @@
+use std::{
+    ops::Shr,
+    simd::{prelude::*, ToBitMask},
+};
+
+macro_rules! low_nibble_mask {
```

Comment:
Needs a comment or it'll be magic to use in a few weeks!

Reply:
I'll move the code to the comment that bleongs to it :D, good call!